gcloud/datastore/_implicit_environ.py

```diff
@@ -79,6 +86,16 @@ def compute_engine_id():
         connection.close()
 
 
+def _get_production_dataset_id():
```

Comment:
I guess I don't see the need to separate them (you have to add tests for them later, for instance).  As they stood, the lookups were obvious to the eye.  The extra function call isn't an issue, performance wise:  it just seems excessive.

Maybe I've lost track:  are you making them separate functions in order to monkey them as functions somewhere, rather than monkeying the environ?